### DISCUSSION TITLE
How to decouple the theme for greater customization?

### DISCUSSION THREAD

**Discussion by jassimahmed** [LekoArts/gatsby-themes#991] [Questions]
Basically I want to cutomize the theme much more than what shadowing allows. I have tried to use the theme as a plugin and local dependency, but neither works. Any suggestions?

**Comment by LekoArts** [comment]
Fork this repository, remove any themes and examples that are not relevant and point your deploy dir to the example then.

**Comment by jassimahmed** [comment]
Thank you. This works on my local machine. But trying to building it in netlify doesn't seem to work. '

I used yarn workspace minimal-blog build

9:14:02 PM: yarn workspaces detected
9:14:02 PM: Installing NPM modules using Yarn version 3.2.3
9:14:03 PM: ➤ YN0050: The cache-folder option has been deprecated; use rc settings instead
9:14:03 PM: ➤ YN0000: ┌ Resolution step
9:14:03 PM: ➤ YN0002: │ @lekoarts/gatsby-theme-minimal-blog@workspace:themes/gatsby-theme-minimal-blog doesn't provide @theme-ui/css (pf8142), requested by gatsby-plugin-theme-ui
➤ YN0002: │ @lekoarts/gatsby-theme-minimal-blog@workspace:themes/gatsby-theme-minimal-blog doesn't provide @types/mdx (paae52), requested by @theme-ui/mdx
➤ YN0002: │ @lekoarts/gatsby-theme-minimal-blog@workspace:themes/gatsby-theme-minimal-blog [9d7ab] doesn't provide @theme-ui/css (p2be3e), requested by gatsby-plugin-theme-ui
➤ YN0002: │ @lekoarts/gatsby-theme-minimal-blog@workspace:themes/gatsby-theme-minimal-blog [9d7ab] doesn't provide @types/mdx (p50b05), requested by @theme-ui/mdx
➤ YN0060: │ @lekoarts/gatsby-themes@workspace:. provides @types/react (p528b1) with version 18.0.21, which doesn't satisfy what netlify-cms-app and some of its descendants request
➤ YN0060: │ @lekoarts/gatsby-themes@workspace:. provides cypress (p57178) with version 10.9.0, which doesn't satisfy what gatsby-cypress requests
➤ YN0002: │ @lekoarts/gatsby-themes@workspace:. doesn't provide gatsby (p668ef), requested by gatsby-cypress
➤ YN0002: │ @lekoarts/gatsby-themes@workspace:. doesn't provide gatsby (pd9b91), requested by gatsby-plugin-netlify-cms
➤ YN0002: │ @lekoarts/gatsby-themes@workspace:. doesn't provide react (p7aa41), requested by @testing-library/react
➤ YN0002: │ @lekoarts/gatsby-themes@workspace:. doesn't provide react (pf709c), requested by gatsby-plugin-netlify-cms
➤ YN0002: │ @lekoarts/gatsby-themes@workspace:. doesn't provide react (p692cd), requested by netlify-cms-app
➤ YN0002: │ @lekoarts/gatsby-themes@workspace:. doesn't provide react-dom (p3dcbb), requested by @testing-library/react
➤ YN0002: │ @lekoarts/gatsby-themes@workspace:. doesn't provide react-dom (pb3e1b), requested by gatsby-plugin-netlify-cms
➤ YN0002: │ @lekoarts/gatsby-themes@workspace:. doesn't provide react-dom (p2357f), requested by netlify-cms-app
➤ YN0002: │ @lekoarts/gatsby-themes@workspace:. doesn't provide webpack (p139e7), requested by gatsby-plugin-netlify-cms
➤ YN0002: │ @parcel/types@npm:2.6.2 doesn't provide @parcel/core (p1835f), requested by @parcel/fs
➤ YN0002: │ @parcel/types@npm:2.6.2 doesn't provide @parcel/core (p40766), requested by @parcel/workers
➤ YN0002: │ @parcel/types@npm:2.6.2 doesn't provide @parcel/core (p60182), requested by @parcel/cache
➤ YN0002: │ @parcel/types@npm:2.6.2 doesn't provide @parcel/core (pe24cb), requested by @parcel/package-manager
➤ YN0002: │ @swc-node/register@npm:1.5.2 [e3223] doesn't provide @swc/core (p8a828), requested by @swc-node/core
➤ YN0002: │ @theme-ui/color@npm:0.15.1 doesn't provide @emotion/react (pe64d5), requested by @theme-ui/css
➤ YN0002: │ @theme-ui/preset-sketchy@npm:0.15.1 doesn't provide @emotion/react (p7cec6), requested by @theme-ui/css
➤ YN0002: │ @theme-ui/presets@npm:0.15.1 doesn't provide @theme-ui/css (p1daa8), requested by @theme-ui/preset-base
➤ YN0002: │ @theme-ui/presets@npm:0.15.1 doesn't provide @theme-ui/css (pd60a3), requested by @theme-ui/preset-bootstrap
➤ YN0002: │ @theme-ui/presets@npm:0.15.1 doesn't provide @theme-ui/css (pbb19b), requested by @theme-ui/preset-bulma
➤ YN0002: │ @theme-ui/presets@npm:0.15.1 doesn't provide @theme-ui/css (pa2a2b), requested by @theme-ui/preset-dark
➤ YN0002: │ @theme-ui/presets@npm:0.15.1 doesn't provide @theme-ui/css (p182f8), requested by @theme-ui/preset-deep
➤ YN0002: │ @theme-ui/presets@npm:0.15.1 doesn't provide @theme-ui/css (p20899), requested by @theme-ui/preset-funk
➤ YN0002: │ @theme-ui/presets@npm:0.15.1 doesn't provide @theme-ui/css (pc070e), requested by @theme-ui/preset-future
➤ YN0002: │ @theme-ui/presets@npm:0.15.1 doesn't provide @theme-ui/css (p9b595), requested by @theme-ui/preset-polaris
➤ YN0002: │ @theme-ui/presets@npm:0.15.1 doesn't provide @theme-ui/css (p412bf), requested by @theme-ui/preset-roboto
➤ YN0002: │ @theme-ui/presets@npm:0.15.1 doesn't provide @theme-ui/css (p615bb), requested by @theme-ui/preset-swiss
➤ YN0002: │ @theme-ui/presets@npm:0.15.1 doesn't provide @theme-ui/css (p2efe8), requested by @theme-ui/preset-system
➤ YN0002: │ @theme-ui/presets@npm:0.15.1 doesn't provide @theme-ui/css (p275d5), requested by @theme-ui/preset-tailwind
➤ YN0002: │ @theme-ui/presets@npm:0.15.1 doesn't provide @theme-ui/css (p8e6b0), requested by @theme-ui/preset-tosh
➤ YN0002: │ apollo-link-context@npm:1.0.20 doesn't provide graphql (pa199c), requested by apollo-link
➤ YN0002: │ gatsby-plugin-feed@npm:4.24.0 [9d7ab] doesn't provide graphql (pc53d5), requested by gatsby-plugin-utils
➤ YN0002: │ gatsby-plugin-manifest@npm:4.24.0 [9d7ab] doesn't provide graphql (p56f15), requested by gatsby-plugin-utils
➤ YN0002: │ gatsby-plugin-mdx@npm:4.3.0 [10fe1] doesn't provide graphql (p27363), requested by gatsby-plugin-utils
➤ YN0002: │ gatsby-plugin-mdx@npm:4.3.0 [7cd17] doesn't provide graphql (p1c6a3), requested by gatsby-plugin-utils
➤ YN0002: │ gatsby-plugin-page-creator@npm:4.24.0 [8902d] doesn't provide graphql (pafdf8), requested by gatsby-plugin-utils
➤ YN0002: │ gatsby-plugin-sharp@npm:4.24.0 [10fe1] doesn't provide graphql (p2f8a9), requested by gatsby-plugin-utils
➤ YN0002: │ gatsby-plugin-sharp@npm:4.24.0 [7cd17] doesn't provide graphql (p15fba), requested by gatsby-plugin-utils
➤ YN0002: │ gatsby-transformer-sharp@npm:4.24.0 [10fe1] doesn't provide graphql (p2b3a2), requested by gatsby-plugin-utils
➤ YN0002: │ gatsby-transformer-sharp@npm:4.24.0 [7cd17] doesn't provide graphql (pfac18), requested by gatsby-plugin-utils
➤ YN0002: │ gatsby@npm:4.24.1 [9d7ab] doesn't provide babel-eslint (p64318), requested by eslint-config-react-app
➤ YN0060: │ minimal-blog@workspace:examples/minimal-blog provides react (p8f074) with version 18.2.0, which doesn't satisfy what @lekoarts/gatsby-theme-minimal-blog and some of its descendants request
➤ YN0060: │ minimal-blog@workspace:examples/minimal-blog provides react (p9a0dc) with version 18.2.0, which doesn't satisfy what gatsby and some of its descendants request
➤ YN0060: │ minimal-blog@workspace:examples/minimal-blog provides react (pc5826) with version 18.2.0, which doesn't satisfy what netlify-cms-app and some of its descendants request
➤ YN0060: │ minimal-blog@workspace:examples/minimal-blog provides react-dom (p3d935) with version 18.2.0, which doesn't satisfy what netlify-cms-app and some of its descendants request
➤ YN0002: │ minimal-blog@workspace:examples/minimal-blog doesn't provide webpack (p98847), requested by gatsby-plugin-gatsby-cloud
➤ YN0002: │ minimal-blog@workspace:examples/minimal-blog doesn't provide webpack (pa1212), requested by gatsby-plugin-netlify-cms
➤ YN0002: │ netlify-cms-backend-gitlab@npm:2.13.0 [0493d] doesn't provide graphql (pef8a6), requested by apollo-cache-inmemory
➤ YN0002: │ netlify-cms-backend-gitlab@npm:2.13.0 [0493d] doesn't provide graphql (p89600), requested by apollo-client
➤ YN0002: │ netlify-cms-backend-gitlab@npm:2.13.0 [0493d] doesn't provide graphql (p9d7b8), requested by apollo-link-http
➤ YN0002: │ netlify-cms-backend-gitlab@npm:2.13.0 [dff08] doesn't provide graphql (pb0716), requested by apollo-cache-inmemory
➤ YN0002: │ netlify-cms-backend-gitlab@npm:2.13.0 [dff08] doesn't provide graphql (pa3d0c), requested by apollo-client
➤ YN0002: │ netlify-cms-backend-gitlab@npm:2.13.0 [dff08] doesn't provide graphql (p201df), requested by apollo-link-http
➤ YN0060: │ netlify-cms-core@npm:2.55.2 [0493d] provides ajv (pc35e2) with version 8.1.0, which doesn't satisfy what ajv-keywords requests
➤ YN0060: │ netlify-cms-core@npm:2.55.2 [dff08] provides ajv (pb4d89) with version 8.1.0, which doesn't satisfy what ajv-keywords requests
➤ YN0002: │ netlify-cms-ui-default@npm:2.15.5 [0493d] doesn't provide react-dom (pbe65d), requested by react-transition-group
➤ YN0002: │ netlify-cms-ui-default@npm:2.15.5 [dff08] doesn't provide react-dom (p4fc2f), requested by react-transition-group
➤ YN0002: │ netlify-cms-widget-code@npm:1.3.4 [0493d] doesn't provide react-dom (p21668), requested by react-select
➤ YN0002: │ netlify-cms-widget-code@npm:1.3.4 [dff08] doesn't provide react-dom (p01550), requested by react-select
➤ YN0002: │ netlify-cms-widget-datetime@npm:2.7.4 [0493d] doesn't provide moment (p98a2e), requested by react-datetime
➤ YN0002: │ netlify-cms-widget-datetime@npm:2.7.4 [dff08] doesn't provide moment (p4e5a1), requested by react-datetime
➤ YN0002: │ netlify-cms-widget-file@npm:2.12.1 [0493d] doesn't provide react-dom (pc12ef), requested by react-sortable-hoc
➤ YN0002: │ netlify-cms-widget-file@npm:2.12.1 [dff08] doesn't provide react-dom (p166aa), requested by react-sortable-hoc
➤ YN0002: │ netlify-cms-widget-list@npm:2.10.1 [0493d] doesn't provide react-dom (p01803), requested by react-sortable-hoc
➤ YN0002: │ netlify-cms-widget-list@npm:2.10.1 [dff08] doesn't provide react-dom (p31f4f), requested by react-sortable-hoc
➤ YN0002: │ netlify-cms-widget-relation@npm:2.11.1 [0493d] doesn't provide react-dom (pfff25), requested by react-select
➤ YN0002: │ netlify-cms-widget-relation@npm:2.11.1 [0493d] doesn't provide react-dom (p4b61c), requested by react-sortable-hoc
➤ YN0002: │ netlify-cms-widget-relation@npm:2.11.1 [0493d] doesn't provide react-dom (pa9616), requested by react-window
➤ YN0002: │ netlify-cms-widget-relation@npm:2.11.1 [dff08] doesn't provide react-dom (p33b3a), requested by react-select
➤ YN0002: │ netlify-cms-widget-relation@npm:2.11.1 [dff08] doesn't provide react-dom (p82141), requested by react-sortable-hoc
➤ YN0002: │ netlify-cms-widget-relation@npm:2.11.1 [dff08] doesn't provide react-dom (p0555e), requested by react-window
➤ YN0002: │ netlify-cms-widget-select@npm:2.8.2 [0493d] doesn't provide react-dom (p75401), requested by react-select
➤ YN0002: │ netlify-cms-widget-select@npm:2.8.2 [dff08] doesn't provide react-dom (p913a0), requested by react-select
➤ YN0002: │ redux-notifications@npm:4.0.1 [103ac] doesn't provide redux (pbe711), requested by react-redux
➤ YN0002: │ redux-notifications@npm:4.0.1 [6e0df] doesn't provide redux (p1097f), requested by react-redux
➤ YN0002: │ theme-ui@npm:0.15.1 [41563] doesn't provide @emotion/react (p7b4fe), requested by @theme-ui/css
➤ YN0002: │ theme-ui@npm:0.15.1 [41563] doesn't provide @emotion/react (p3303f), requested by @theme-ui/core
➤ YN0002: │ theme-ui@npm:0.15.1 [41563] doesn't provide @emotion/react (p395db), requested by @theme-ui/color-modes
➤ YN0002: │ theme-ui@npm:0.15.1 [41563] doesn't provide @emotion/react (p98166), requested by @theme-ui/components
➤ YN0002: │ theme-ui@npm:0.15.1 [41563] doesn't provide @emotion/react (p8c25e), requested by @theme-ui/theme-provider
➤ YN0002: │ theme-ui@npm:0.15.1 [869d1] doesn't provide @emotion/react (p7df85), requested by @theme-ui/css
➤ YN0002: │ theme-ui@npm:0.15.1 [869d1] doesn't provide @emotion/react (p79aff), requested by @theme-ui/core
➤ YN0002: │ theme-ui@npm:0.15.1 [869d1] doesn't provide @emotion/react (pf19e3), requested by @theme-ui/color-modes
➤ YN0002: │ theme-ui@npm:0.15.1 [869d1] doesn't provide @emotion/react (p8ea08), requested by @theme-ui/components
➤ YN0002: │ theme-ui@npm:0.15.1 [869d1] doesn't provide @emotion/react (pe79fc), requested by @theme-ui/theme-provider
➤ YN0000: │ Some peer dependencies are incorrectly met; run yarn explain peer-requirements <hash> for details, where <hash> is the six-letter p-prefixed code
9:14:03 PM: ➤ YN0000: └ Completed in 0s 368ms
9:14:03 PM: ➤ YN0000: ┌ Fetch step
9:15:02 PM: ➤ YN0066: │ typescript@patch:typescript@npm%3A4.8.4#~builtin<compat/typescript>::version=4.8.4&hash=a1c5e5: Cannot apply hunk #7
9:15:15 PM: ➤ YN0013: │ 72 packages were already cached, 2280 had to be fetched
9:15:15 PM: ➤ YN0000: └ Completed in 1m 12s
9:15:15 PM: ➤ YN0000: ┌ Link step
9:15:19 PM: ➤ YN0000: │ ESM support for PnP uses the experimental loader API and is therefore experimental
9:15:20 PM: ➤ YN0007: │ @swc/core@npm:1.3.4 must be built because it never has been before or the last one failed
9:15:20 PM: ➤ YN0007: │ esbuild@npm:0.15.7 must be built because it never has been before or the last one failed
9:15:20 PM: ➤ YN0007: │ cypress@npm:10.9.0 must be built because it never has been before or the last one failed
9:15:20 PM: ➤ YN0007: │ core-js-pure@npm:3.25.1 must be built because it never has been before or the last one failed
9:15:20 PM: ➤ YN0007: │ esbuild@npm:0.15.10 must be built because it never has been before or the last one failed
9:15:20 PM: ➤ YN0007: │ lmdb@npm:2.5.3 must be built because it never has been before or the last one failed
9:15:20 PM: ➤ YN0007: │ msgpackr-extract@npm:2.1.2 must be built because it never has been before or the last one failed
9:15:20 PM: ➤ YN0007: │ sharp@npm:0.30.7 must be built because it never has been before or the last one failed
9:15:20 PM: ➤ YN0007: │ gatsby-telemetry@npm:3.24.0 must be built because it never has been before or the last one failed
9:15:20 PM: ➤ YN0007: │ lmdb@npm:2.5.2 must be built because it never has been before or the last one failed
9:15:20 PM: ➤ YN0007: │ @parcel/watcher@npm:2.0.5 must be built because it never has been before or the last one failed
9:15:20 PM: ➤ YN0007: │ core-js@npm:3.25.1 must be built because it never has been before or the last one failed
9:15:20 PM: ➤ YN0007: │ gatsby-cli@npm:4.24.0 must be built because it never has been before or the last one failed
9:15:20 PM: ➤ YN0007: │ es5-ext@npm:0.10.62 must be built because it never has been before or the last one failed
9:15:21 PM: ➤ YN0000: │ cypress@npm:10.9.0 STDOUT Note: Overriding Cypress cache directory to: ./node_modules/.cache/CypressBinary
9:15:21 PM: ➤ YN0000: │ cypress@npm:10.9.0 STDOUT
9:15:21 PM: ➤ YN0000: │ cypress@npm:10.9.0 STDOUT       Previous installs of Cypress may not be found.
9:15:21 PM: ➤ YN0000: │ cypress@npm:10.9.0 STDOUT
9:15:21 PM: ➤ YN0000: │ cypress@npm:10.9.0 STDOUT Installing Cypress (version: 10.9.0)
9:15:21 PM: ➤ YN0000: │ cypress@npm:10.9.0 STDOUT
9:15:21 PM: ➤ YN0000: │ cypress@npm:10.9.0 STDOUT [STARTED] Task without title.
9:15:22 PM: ➤ YN0000: │ sharp@npm:0.30.7 STDOUT sharp: Downloading https://github.com/lovell/sharp-libvips/releases/download/v8.12.2/libvips-8.12.2-linux-x64.tar.br
9:15:23 PM: ➤ YN0000: │ sharp@npm:0.30.7 STDOUT sharp: Integrity check passed for linux-x64
9:15:24 PM: ➤ YN0000: │ cypress@npm:10.9.0 STDOUT [SUCCESS] Task without title.
9:15:24 PM: ➤ YN0000: │ cypress@npm:10.9.0 STDOUT [STARTED] Task without title.
9:15:28 PM: ➤ YN0000: │ cypress@npm:10.9.0 STDOUT [SUCCESS] Task without title.
9:15:28 PM: ➤ YN0000: │ cypress@npm:10.9.0 STDOUT [STARTED] Task without title.
9:15:28 PM: ➤ YN0000: │ cypress@npm:10.9.0 STDOUT [SUCCESS] Task without title.
9:15:29 PM: ➤ YN0000: │ cypress@npm:10.9.0 STDOUT
9:15:29 PM: ➤ YN0000: │ cypress@npm:10.9.0 STDOUT You can now open Cypress by running: node_modules/.bin/cypress open
9:15:29 PM: ➤ YN0000: │ cypress@npm:10.9.0 STDOUT
9:15:29 PM: ➤ YN0000: │ cypress@npm:10.9.0 STDOUT https://on.cypress.io/installing-cypress
9:15:29 PM: ➤ YN0000: │ cypress@npm:10.9.0 STDOUT
9:15:29 PM: ➤ YN0007: │ @lekoarts/gatsby-themes@workspace:. must be built because it never has been before or the last one failed
9:15:30 PM: ➤ YN0007: │ gatsby@npm:4.24.1 [9d7ab] must be built because it never has been before or the last one failed
9:15:31 PM: ➤ YN0000: │ @lekoarts/gatsby-themes@workspace:. STDOUT Unknown Syntax Error: Command not found; did you mean:
9:15:31 PM: ➤ YN0000: │ @lekoarts/gatsby-themes@workspace:. STDOUT
9:15:31 PM: ➤ YN0000: │ @lekoarts/gatsby-themes@workspace:. STDOUT $ yarn workspaces list [--since] [-R,--recursive] [-v,--verbose] [--json]
9:15:31 PM: ➤ YN0000: │ @lekoarts/gatsby-themes@workspace:. STDOUT While running workspaces foreach --parallel --interlaced run package:build
9:15:31 PM: ➤ YN0009: │ @lekoarts/gatsby-themes@workspace:. couldn't be built successfully (exit code 1, logs can be found here: /tmp/xfs-cdc81108/build.log)
➤ YN0000: └ Completed in 15s 565ms
9:15:31 PM: ➤ YN0000: Failed with errors in 1m 29s
9:15:32 PM: Error during Yarn install
9:15:32 PM: Build was terminated: Build script returned non-zero exit code: 1
9:15:32 PM: Creating deploy upload records
9:15:32 PM: Failing build: Failed to build site
9:15:32 PM: Failed during stage 'building site': Build script returned non-zero exit code: 1 (https://ntl.fyi/exit-code-1)
9:15:32 PM: Finished processing build request in 1m38.434330003s
